third_party/amd/lib/TritonAMDGPUTransforms/Utility.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Utility.h"
+
+#include "mlir/Dialect/SCF/IR/SCF.h"
+
+namespace findMin {
+int findMinInBlock(Block &block,
+                   const std::function<int(Operation *)> &countFunc);
+
+// Returns the minimum found when accumulating countFunc(op) for all paths
+// between op1 and op2
+int findMinBetweenOps(Operation *op1, Operation *op2,
+                      const std::function<int(Operation *)> &countFunc) {
+  assert(op1 && op2);
```

Comment:
Need to also assert `op1-> isBeforeInBlock(op2)` otherwise we have an infinite loop at L15?